src/main.c

```diff
@@ -0,0 +1,210 @@
+    int32_t num2 = 0;                           /**< Variable that will hold an operand with MAX 32 bits */
+    int32_t result = 0;                         /**< Variable that will hold the resulting calculated number */
+    char *numEnd1 = NULL;                       /**< Pointer to the end of num1 */
+    char *numEnd2 = NULL;                       /**< Pointer to the end of num2 */                               
+    
+    Operation operation;                        
+    errno = 0;                                  /**< Reset errno */
+
+    /**
+    * @brief Error-checking for number of CMD Line arguments
+    * Should be 4 <filename> <operand> <operator> <operand>
+    */
+
+    if (ARGCOUNT != argc)                                  
+    {
+        fprintf(stderr, "Argument Error! Reading %d of %d required args\n", argc, ARGCOUNT);
+        // Handle wildcard expansion / globbing 
+        if (argc > ARGCOUNT)            
+        {
+            fprintf(stderr, "Likely encountered wildcard expansion. Precede CLI command with commmand 'set -f'\n");
+        } 
+        printf("See below for more troubleshooting\n");
+        printHelp(argv[0]);
+        exit(EXIT_FAILURE);
+    }
+    else
+        argCheck = true;
+    
+/**
+ * @brief Error-checking for valid first operand <num1> 
+ * 
+ */
+
+    num1 = strtol(argv[1], &numEnd1, 10);
+    if (0 != errno || '\0' != *numEnd1)     /**< Check for over/underflow and validity of string */
+    {
+        fprintf(stderr, "'%s' is an invalid operand\n", argv[1]);
+        printf("Make sure you input the following:\n");
+        printHelp(argv[0]);
+        exit(EXIT_FAILURE);
+    }
+    else
+        hasNum1 = true;
+
+/**
+ * @brief Error-checking for valid first operand <num1> 
+ * 
+ */
+
+    num2 = strtol(argv[3], &numEnd2, 10);
+    if (0 != errno || '\0' != *numEnd2)     /**< Check for over/underflow and validity of string */
+    {
+        fprintf(stderr, "'%s' is an invalid operand\n", argv[3]);
+        printf("Make sure you input the following:\n");
+        printHelp(argv[0]);
+        exit(EXIT_FAILURE);
+    }
+    else
+        hasNum2 = true;
+
+/**
+ * @brief Error-checking for valid arithmetic operator 
+ * 
+ */
+
+    operation = checkOperator(argv[2]);
+
+    if (operation == INVALID)
+    {
+        fprintf(stderr, "'%s' is an invalid operator\n", argv[2]);
+        printf("Make sure you input the following:\n");
+        printHelp(argv[0]);
+        exit(EXIT_FAILURE);
+    }
+    else
+        opCheck = true;
+
+/**
+ * @brief Based on passing error-checks, execute the proper arithmetic
+ * operation and print the results in the form of <num1> <operator> <num2> = <result> 
+ * to the terminal.
+ * 
+ * All the cases for the switch (operation) refer to the enumeration defined at the beginning
+ * of the code. 
+ */
+
+    if (argCheck && hasNum1 && hasNum2 && opCheck)
```

Comment:
Make all explicit 

```suggestion
    if ((true == argCheck) && (true == hasNum1) && (true == hasNum2) && (true == opCheck)) 
```

Can also simplify by anding all Boolean's beforehand (only enter condition if all are true)